app/domain/service/retry.go

```diff
@@ -57,33 +38,28 @@ func timeout() <-chan retry.Result {
 // the query is stuck forever and breaks the business logic. This way, if the query takes more than sleepPeriod, it will
 // retry the query {@param retries} times.
 // If {@param retries} is reached, it will return an error.
-func Retry(executionFunction func() <-chan retry.Result, retries int) (interface{}, error) {
+func Retry(executionFunction func(context.Context) retry.Result, retries int) (interface{}, error) {
 	times := 0
-	var retryFunction func() (interface{}, error)
 
-	retryFunction = func() (interface{}, error) {
-		var executionResult retry.Result
-		select {
-		case executionResult = <-timeout():
-		case executionResult = <-executionFunction():
-		}
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), sleepPeriod)
+		executionResult := executionFunction(ctx)
+		cancel()
```

Comment:
defer cancel()?

Later, if someone uses ctx, it will be canceled and the request could not be fired